engine/src/main/java/org/kigalisim/engine/support/EolEmissionsRecalcStrategy.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Strategy for recalculating end-of-life emissions.
+ *
+ * <p>This strategy encapsulates the logic previously found in the
+ * recalcEolEmissions method of SingleThreadEngine.</p>
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.support;
+
+import org.kigalisim.engine.Engine;
+import org.kigalisim.engine.SingleThreadEngine;
+import org.kigalisim.engine.state.Scope;
+
+/**
+ * Strategy for recalculating end-of-life emissions.
+ */
+public class EolEmissionsRecalcStrategy implements RecalcStrategy {
+
+  private Scope scope;
+
+  /**
+   * Create a new EolEmissionsRecalcStrategy.
+   *
+   * @param scope The scope to use for calculations, null to use engine's current scope
+   */
+  public EolEmissionsRecalcStrategy(Scope scope) {
+    this.scope = scope;
+  }
+
+  @Override
+  public void execute(Engine target) {
+    if (!(target instanceof SingleThreadEngine)) {
```

Comment:
Please evaluate if this is still necessary after the other changes or if `Engine` can be used directly.